### ISSUE TITLE
dyld error with globally installed libvps on macos

### ISSUE THREAD

**Issue by nirbhasa** [lovell/sharp#3528]
I am using sharp inside an electronjs app, macos 13.0, x64.

When I install libvips globally using homebrew, and then install sharp, my electronjs app crashes when sharp is utilised (the window just turns completely white), with the following message `dyld[74585]: missing symbol called`

When I uninstall libvips globally and then uninstall/reinstall sharp, everything is fine.

**Comment by lovell** [comment]
There's not a lot to go on here, but please check you're using the latest versions of both Electron and sharp. Please also check that *all* of your brew-installed dependencies are latest (using something like `brew update && brew upgrade`).

* Does `vips` work at the command line?
* Does sharp linked against the global libvips work with Node.js (i.e. without Electron)?
* Is this an Intel or ARM machine? Are you definitely using the same architecture everywhere?

**Comment by nirbhasa** [comment]
Thanks for getting back to me! So....

- Updated Electron and all brew dependencies: error persists
- vips works at command line
- sharp linked against the global libvips works with Node.js (without Electron) 
- intel machine

Ill try running sharp (linked against the global libvips) in barebones electronjs app and see how i get on...

**Comment by lovell** [comment]
@nirbhasa Were you able to make any progress with this?

**Comment by lovell** [comment]
Closing due to inactivity but please feel free to reopen with more details if further help is required.

**Comment by nirbhasa** [comment]
Certainly – thanks for following up. I had to push this to backburner but hopefully I can look at it again at some point

**Comment by sgerace** [comment]
I'm seeing this same (or very similar) issue on my machine as well. It certainly seems to be related to `electron-builder` in my case. Here's the scenario I'm running into:

If I uninstall `vips` globally (with `brew uninstall vips`) and clear out my `node_modules` folder, then run `npm install`, everything works — besides the fact that I don't have HEIC support, which is why I need to compile against a global version of vips. 

If I then install vips globally (with `brew install vips`), and clear out my `node_modules` again, and run `npm install` (which runs `electron-builder` as a post-install step), I get the following error when attempting to run a script that utilizes sharp:

```
dyld: lazy symbol binding failed: Symbol not found: _g_once_impl
  Referenced from: /Users/dev/Projects/smp/node_modules/sharp/build/Release/sharp-darwin-x64.node
  Expected in: /usr/local/lib/libvips-cpp.42.dylib

dyld: Symbol not found: _g_once_impl
  Referenced from: /Users/dev/Projects/smp/node_modules/sharp/build/Release/sharp-darwin-x64.node
  Expected in: /usr/local/lib/libvips-cpp.42.dylib

Abort trap: 6
```

Interestingly, I found a related comment: https://github.com/lovell/sharp/issues/2966#issuecomment-962585153 in a different issue that indicates `electron-builder` might not play nicely with sharp in some cases. So, to test this theory, I ran   `rm -rf node_modules/sharp && npm install sharp` _after_ running `npm install` (i.e., _after_ `electron-builder` has run). At this point, my script is running properly without any errors and WITH HEIC support. 

Unfortunately, this "workaround" (`rm -rf node_modules/sharp && npm install sharp` after `npm install`) only seems to work on my development environment. I can't find any way to prevent `electron-builder` from running when I ultimately run `electron-builder` to create the distributable Electron package.

Hopefully this helps track down the root cause. Fortunately, I _can_ perform development locally, however, as I'm unable to build any production artifacts, this is a bit of a blocker for me at the moment.

Thanks!
